content/en/docs/guides/debugging.md

```diff
@@ -0,0 +1,45 @@
+---
+title: Debugging
+description: >-
+  Explains the debugging process of gRPC applications using [grpcdebug]
+---
+
+### Overview
+[grpcdebug] is a command line tool within the gRPC ecosystem designed to assist developers in debugging and troubleshooting gRPC services. grpcdebug fetches the internal states of the gRPC library from the application via gRPC protocol and provide a human-friendly UX to browse them. Currently, it supports [Channelz]/[Health] Checking/CSDS (aka. [admin services]). In other words, it can fetch statistics about how many RPCs has being sent or failed on a given gRPC channel, it can inspect address resolution results, it can dump the in-effective xDS configuration that directs the routing of RPCs.
+
+### Setup
```

Comment:
How do you feel about replacing the "Setup", "Usage" and "Getting Started" sections with "Examples" and "References"?

Under examples you can mention that different language implementations provide example service implementations and instructions on how to debug them and then provide links to them. E.g. the Java one is here: https://github.com/grpc/grpc-java/tree/master/examples/example-debug

Under references you could provide these other links you have at the bottom?

I don't think we need to copy over the usage details of the tool as they might just get out of date and we don't want this to be the main reference for the tool.

What do you think?

Reply:
or it might just be better to have a general debugging guide linking to different languages debugging sections and each language will have grpcdebug section?

Reply:
For now I have kept the grpcdebug part but as per your suggestion removed everything else and linked the examples from different languages. For go, I have linked grpcdebug doc because they implemented the server in Go for showcasing the commands